source/Plugins/BattleNetLibraryCore/BattleNetGameController.cs

```diff
@@ -0,0 +1,226 @@
+﻿using BattleNetLibrary.Models;
+using Playnite;
+using Playnite.Common;
+using Playnite.SDK;
+using Playnite.SDK.Events;
+using Playnite.SDK.Models;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BattleNetLibrary
+{
+    public class BattleNetGameController : BaseGameController
+    {
+        private static ILogger logger = LogManager.GetLogger();
+        private CancellationTokenSource watcherToken;
+        private ProcessMonitor procMon;
+        private Stopwatch stopWatch;
+        private IPlayniteAPI api;
+
+        public BattleNetGameController(Game game, IPlayniteAPI api) : base(game)
+        {
+            this.api = api;
+        }
+
+        public override void Dispose()
+        {
+            ReleaseResources();
+        }
+
+        public void ReleaseResources()
+        {
+            procMon?.Dispose();
```

Comment:
should detach from the monitor_treedestroyed as well?

Reply:
It shouldn't be necessary, as the source of the event is being disposed, the subscriptions are removed. But it's always a good practice